File: ts-sdk/src/sdk/models/operations/postoauth2tokennumberrefresh.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";

export const PostOauth2TokenNumberRefreshServerList = [
	"https://api.box.com",
] as const;


export class PostOauth2TokenNumberRefreshRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  serverURL?: string;

  @SpeakeasyMetadata({ data: "request, media_type=application/x-www-form-urlencoded" })
  request?: shared.PostOAuth2TokenRefreshAccessToken;
}


export class PostOauth2TokenNumberRefreshResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  accessToken?: shared.AccessToken;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  oAuth2Error?: shared.OAuth2Error;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-sdk/pkg/models/operations/postoauth2tokennumberrefresh.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

var PostOauth2TokenNumberRefreshServerList = []string{
	"https://api.box.com",
}

type PostOauth2TokenNumberRefreshRequest struct {
	Retries   *utils.RetryConfig
	ServerURL *string
	Request   *shared.PostOAuth2TokenRefreshAccessToken `request:"mediaType=application/x-www-form-urlencoded"`
}

type PostOauth2TokenNumberRefreshResponse struct {
	AccessToken *shared.AccessToken
	ContentType string
	OAuth2Error *shared.OAuth2Error
	StatusCode  int64
}
